src/pages/Kavita/main.ts

```diff
@@ -0,0 +1,200 @@
+      return seriesDetails.seriesID;
+    },
+    getOverviewUrl(url) {
+      return window.location.href.split('/').slice(0, 7).join('/');
+    },
+    getEpisode(url) {
+      if (!chapterDetails.chapterNum || !parseInt(chapterDetails.chapterNum))
+        throw 'No chapter number';
+      return parseInt(chapterDetails.chapterNum);
+    },
+    getVolume(url) {
+      return parseInt(chapterDetails.volumeNum);
+    },
+    getMalUrl(provider) {
+      if (provider === 'MAL' && seriesDetails.malLink) return seriesDetails.malLink;
+      if (provider === 'ANILIST' && seriesDetails.anilistLink) return seriesDetails.anilistLink;
+      if (provider === 'KITSU' && seriesDetails.kitsuLink) return seriesDetails.kitsuLink;
+      return false;
+    },
+    readerConfig: [
+      {
+        current: {
+          callback: () => chapterDetails.currentPageNum as any,
+          mode: 'callback',
+        },
+        total: {
+          callback: () => chapterDetails.totalPageNum as any,
+          mode: 'callback',
+        },
+      },
+    ],
+  },
+  overview: {
+    getTitle() {
+      if (!seriesDetails.seriesTitle) throw 'No name';
+      return seriesDetails.seriesTitle;
+    },
+    getIdentifier(url) {
+      if (!seriesDetails.seriesID) throw 'No series id';
+      return seriesDetails.seriesID;
+    },
+    uiSelector(selector) {
+      return j.$('h4').first().after(j.html(selector));
+    },
+  },
+  init(page) {
+    api.storage.addStyle(
+      require('!to-string-loader!css-loader!less-loader!./style.less').toString(),
+    );
+
+    let progressUpdater;
+    utils.changeDetect(
+      () => {
+        check();
+      },
+      () => {
+        return utils.urlStrip(window.location.href);
+      },
+    );
+    waitForPageLoad(check);
+
+    async function getSeriesDetails() {
+      const seriesID = window.location.pathname.split('/')[4];
+
+      const url = `${window.location.origin}/api/series/${seriesID}`;
+      const response = await apiCall(url);
+      const responseJSON = JSON.parse(response.responseText);
+      con.log('Series Details Retrieved', responseJSON);
+
+      seriesDetails.seriesTitle = responseJSON.name;
+      seriesDetails.seriesID = seriesID;
+      seriesDetails.totalPageNum = responseJSON.pages;
+    }
+
+    async function getChapterDetails() {
+      const chapterID = window.location.pathname.split('/')[6];
+
+      const url = `${window.location.origin}/api/reader/chapter-info?chapterid=${chapterID}`;
+      const response = await apiCall(url);
+      const responseJSON = JSON.parse(response.responseText);
+      con.log('Chapter Info Retrieved', response);
+
+      chapterDetails.volumeNum = responseJSON.volumeNumber;
+      chapterDetails.chapterNum = responseJSON.chapterNumber < 0 ? 0 : responseJSON.chapterNumber;
+      chapterDetails.totalPageNum = responseJSON.pages;
+    }
+
+    function waitForPageLoad(callback) {
+      utils.waitUntilTrue(
```

Comment:
A bit dagerous. If the script loads late then it could miss it and then nothing happens.

It is porbably better to specially wait for that data. Like directly checking the title etc.

Reply:
Ah that's true. The problem is that I added this because the mal link I'm searching for might not exist, so I can't just wait for that element.

I think I'll add a timeout to that page load, so if the spinner is missed, it'll still proceed with handling the page after 0.5s.